server/controllers/report.controller.js

```diff
@@ -128,22 +128,40 @@ exports.createChartReportPDF = async (req, res) => {
         });
 }
 
-exports.fetchChartReportPDF = async (req, res) => {
-    if (!req.query.reportid)
+exports.createPerformanceReportPDF = async (req, res) => {
+    if (!req.body.reportId)
+        return res.status(400).send({ message: "Content cannot be empty." });
+
+    await reportService.createPerformanceReportPDFByPerformanceReportId(req.body.reportId)
+        .then(response => {
+            if (response) {
+                return res.status(200).send(response);
+            }
+            return res.status(500).send({ message: "The PDF could not be created." });
+        })
+        .catch(err => {
+            return res.status(err.status || 500).send({ message: err.message || "Malfunction in the B&C Engine." });
+        });
+}
+
+exports.fetchReportPDF = async (req, res) => {
+    if (!req.query.chartReportId && !req.query.performanceReportId)
         return res.status(400).send({ message: "Content cannot be empty." });
 
     // create file path
     let filePath;
+    let filePrefix = req.query.chartReportId === undefined ? 'performanceReport' : 'chartReport'
+    let fileId = req.query.chartReportId === undefined ? req.query.performanceReportId : req.query.chartReportId
     if (__dirname !== '/home/runner/work/BC-Engine/BC-Engine/server/controllers') {
-        filePath = `${__dirname.replace("controllers", "")}docs\\pdf_files\\chartReport-${req.query.reportid}.pdf`;
+        filePath = `${__dirname.replace("controllers", "")}docs\\pdf_files\\${filePrefix}-${fileId}.pdf`;
     }
     else {
-        filePath = `${__dirname.replace("controllers", "")}docs/pdf_files/chartReport-${req.query.reportid}.pdf`;
+        filePath = `${__dirname.replace("controllers", "")}docs/pdf_files/${filePrefix}-${fileId}.pdf`;
     }
 
     await res.sendFile(filePath, {}, (err) => {
-        if(err) {
-            return res.status(err.status).send({ message: err.message });
+        if (err) {
+            return res.status(err.status || 500).send({ message: err.message || "File not found." });
```

Comment:
You don't need it since the only error that will occur is that the file doesn't exist and the message will be defined as "File not found" and it's the same story for the status. Also, it is not testable.

Reply:
This is something that you put. I will remove if you're sure

Reply:
it is removed in new versions of main

Reply:
fixed here: https://github.com/BC-Engine-Team/BC-Engine/pull/49/files#diff-ff3afc4586b3e274469d0abc88f5664121618aa9fd1ee7c171b0be6e16e427d0